storage/innobase/fil/fil0crypt.cc

```diff
@@ -1074,7 +1074,52 @@ static void fil_crypt_write_crypt_data_to_page0(fil_space_t *space) {
   mtr.commit();
 }
 
-bool fil_crypt_exclude_tablespace_from_rotation(fil_space_t *space) {
+bool fil_crypt_exclude_tablespace_from_rotation_temporarily(
+    fil_space_t *space) {
+  // We acquire fil_crypt_threads_mutex to stop encryption threads from
+  // generating crypt_data for tablespaces that they are about to encrypt.
+  // Generating crypt_data is the fist step in encryption process.
+  mutex_enter(&fil_crypt_threads_mutex);
+
+  if (space->crypt_data == nullptr) {
+    space->exclude_from_rotation = true;
+    mutex_exit(&fil_crypt_threads_mutex);
+    return true;
+  }
+
+  mutex_exit(&fil_crypt_threads_mutex);
+
+  // crypt_data already existed.
+  fil_space_crypt_t *crypt_data = space->crypt_data;
```

Comment:
Decryption won't remove the crypt_data? Or encryption threads never create a new one?

Because at this point we don't hold any mutexes, I'm wondering if this pointer is guaranteed to be valid / current.

Reply:
encryption threads never remove crypt_data. If it is created once it stays. That's why it is safe to acquire this lock here.